### DISCUSSION TITLE
MatchOrder when auction ends

### DISCUSSION THREAD

**Discussion by Dashon** [ProjectOpenSea/seaport-js#391] [General]
I have an auction order:
`{
        "parameters": {
            "offerer": "0xaBd39d459xxx947444C11x9EeC492059B...",
            "zone": "0x0000000000000000000000000000000000000000",
            "zoneHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
            "startTime": "1698227940",
            "endTime": "1698838140",
            "orderType": 0,
            "offer": [
                {
                    "itemType": 3,
                    "token": "0xxxx1c90311bafxxx839bxxxde0c5c068971xxx",
                    "identifierOrCriteria": "21",
                    "startAmount": "1",
                    "endAmount": "1"
                }
            ],
            "consideration": [
                {
                    "itemType": 1,
                    "token": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
                    "identifierOrCriteria": "0",
                    "startAmount": "3000000000000000",
                    "endAmount": "3000000000000000",
                    "recipient": "0xaBd39d4596dd947444C11b9EeC492059B5c927ed"
                }
            ],
            "totalOriginalConsiderationItems": 1,
            "salt": "0x00000000ffc90d33ee65e567",
            "conduitKey": "0x0000000000000000000000000000000000000000000000000000000000000000",
            "counter": "0"
        },
        "signature": "0xbd8ac37bc6f0101a94ed0f0eba58aa33260bedeb761f3e13982eca416bd3157aa35d4cbbcf24b7e839ae9d72046142297389fbb79c26e82c77a11deeeb2af951"
    }`

and an Offer order:

` {
        "parameters": {
            "offerer": "0xF99b0xxx9eBc11fCc16B1257f04...",
            "zone": "0x0000000000000000000000000000000000000000",
            "zoneHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
            "startTime": "1698228021910",
            "endTime": "1700823621910",
            "orderType": 1,
            "offer": [
                {
                    "itemType": 1,
                    "token": "0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6",
                    "identifierOrCriteria": "0",
                    "startAmount": "3000000000000000",
                    "endAmount": "3000000000000000"
                }
            ],
            "consideration": [
                {
                    "itemType": 3,
                    "token": "0xxxx1c90311bafxxx839bxxxde0c5c068971xxx",
                    "identifierOrCriteria": "21",
                    "startAmount": "1",
                    "endAmount": "1",
                    "recipient": "0xF99b0xxx9eBc11fCc16B1257f04..."
                }
            ],
            "totalOriginalConsiderationItems": 1,
            "salt": "0x000000000....",
            "conduitKey": "0x0000000000000000000000000000000000000000000000000000000000000000",
            "counter": "0"
        },
        "signature": "0x..."
    }`

im trying to fulfill them on behalf of the seller using the MatchOrder method provided in this Library:

`const transaction = await seaport!
    .matchOrders({
      orders: [auctionObj, bidObj],
      fulfillments: [
        {
          offerComponents: [{ orderIndex: 0, itemIndex: 0 }],
          considerationComponents: [{ orderIndex: 1, itemIndex: 0 }],
        },
      ],
    })
    .transact();
`

 I keep getting the error:
`cannot estimate gas; transaction may fail or may require manual gas limit (error={\"reason\":\"cannot estimate gas; transaction may fail or may require manual gas limit\",\"code\":\"UNPREDICTABLE_GAS_LIMIT\",\"error\":{\"reason\":\"processing response error\",\"code\":\"SERVER_ERROR\",\"body\":\"{\\\"jsonrpc\\\":\\\"2.0\\\",\\\"id\\\":49,\\\"error\\\":{\\\"code\\\":3,\\\"message\\\":\\\"execution reverted\\\"`

I've experimented with various configurations for the fulfillments array, but it appears I may be misinterpreting the core functionality of MatchOrder.

**Comment by ryanio** [comment]
just to be clear, the first order you specified is a normal "buy now" listing, it is not an "auction" (there is no real auction functionality in Seaport itself. on OpenSea for auctions the zone is set to an EOA that has the sole authority to execute the fulfillment, e.g. if the auction is successful)

~~the problem is your fulfillments are not complete. you need to match all offer and consideration items from both orders together. for the simple "buy now" example you have here, see [`defaultBuyNowMirrorFulfillment`](https://github.com/ProjectOpenSea/seaport/blob/f1749fd83e368360241b154c8c05637774e9f382/test/utils/encoding.ts#L356-L363) in the seaport tests.~~ actually your fulfillments seem okay for one consideration item, see if setting the gasLimit in the tip below can help return a more helpful error. since the conduitKey is zero, approvals have to be directly to the Seaport contract

to help debug the "execution reverted" error to get an actual error message from the Seaport contract, specify a manual `gasLimit` in the `overrides` for 1,000,000 gas so it doesn't return UNPREDICTABLE_GAS_LIMIT

**Reply by Dashon** [reply]
Thank you for your previous assistance. Setting the gasLimit was extremely helpful as it allowed me to identify the actual error from the Seaport contract. The initial error was due to the invalid timestamp on my offer, and the subsequent error was caused by the ConsiderationNotMet. Your assumption was correct; my fulfillments were incomplete. Thank you again for your help and guidance.

Here is my updated matchOrder request.
` const transaction = await seaport!
    .matchOrders({
      orders: [auctionObj, bidObj],
      fulfillments: [
        {
          offerComponents: [{ orderIndex: 0, itemIndex: 0 }],
          considerationComponents: [{ orderIndex: 1, itemIndex: 0 }],
        },
        {
          offerComponents: [{ orderIndex: 1, itemIndex: 0 }],
          considerationComponents: [{ orderIndex: 0, itemIndex: 0 }],
        },
      ],
    })`


I am setting the conduit to zero. Is there any security risk involved in this? Is the Conduit where I should define my business logic such as "auction not over" or "offer not high enough"?
